### PR TITLE
Cache e2e image builds

### PR DESCRIPTION
**Pull request by alexander-demicev**
**What this PR does / why we need it**:

Cache e2e image builds

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits into logical changes
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

**Review by salasberryfin — DISMISSED** [review]
Thanks for this @alexander-demicev. Perhaps this may also help with adding RKE2 scenarios to short tests.

File: Makefile [hunk 1]
```diff
@@ -349,7 +351,7 @@ docker-pull-prerequisites:
 docker-build-etcdrestore: buildx-machine docker-pull-prerequisites ## Build docker image for a specific architecture
 ## reads Dockerfile from stdin to avoid an incorrectly cached Dockerfile (https://github.com/moby/buildkit/issues/1368)
 	# buildx does not support using local registry for multi-architecture images
-	cat $(EXP_ETCDRESTORE_DIR)/Dockerfile | DOCKER_BUILDKIT=1 BUILDX_BUILDER=$(MACHINE) docker buildx build \
+	cat $(EXP_ETCDRESTORE_DIR)/Dockerfile | DOCKER_BUILDKIT=1 BUILDX_BUILDER=$(MACHINE) docker buildx build $(ADDITIONAL_COMMANDS) \
```
Comment (furkatgofurov7) [inline Makefile:354]: Non-blocking question: there is no need to pass `ADDITIONAL_COMMANDS=$(CACHE_COMMANDS)` when using docker buildx?

**Comment by furkatgofurov7** [comment]
> I'm now seeing some authentication errors when it tries to push the cache data to ghcr registry:
> 
> ```
> ERROR: error writing layer blob: unexpected status from POST request to https://ghcr.io/v2/rancher/turtles-e2e-cache/blobs/uploads/: 403 Forbidden
> ```

Yes, I also noticed it in the e2e tests here

**Comment by alexander-demicev** [comment]
@salasberryfin @furkatgofurov7 github uses my account crendentials to push cache to GCHR registry when running tests from the PR, I changed that to only push cache for nightly builds

File: Makefile (hunk 1 above)
Comment (alexander-demicev) [reply Makefile:354]: it's optional

**Review by salasberryfin — APPROVED** [review]
Thanks, I think it's nice that we can run it on PRs also.

**Review by Danil-Grigorev — APPROVED** [review]

**Pull request merged**